lib/bolt/transport/ssh.rb

```diff
@@ -187,10 +189,15 @@ def run_task(target, task, arguments, options = {})
         end
       end
 
-      def make_wrapper_stringio(task_path, stdin)
+      def make_wrapper_stringio(task_path, stdin, interpreter = nil)
         StringIO.new(<<-SCRIPT)
 #!/bin/sh
-'#{task_path}' <<'EOF'
+if test -z '#{interpreter}'; then
+  set -- '#{task_path}'
```

Comment:
What does `set --` do here?

Reply:
`set --` would unset any positional parameters given to the script, then on the same line replaces them with value for `interpreter` and `task_path`. This allows an array like behavior when executing with `"$@"`. I think that as you mentioned before keeping this wrapper script as simple and straight forward will help keep in maintainable and easy to debug. Updated to simplify.